source/tutorials/ad-hoc-developer-environments.rst

```diff
@@ -0,0 +1,137 @@
+Ad hoc developer environments
+=============================
+
+Assuming you have :ref:`Nix installed <install-nix>`, it is able to download packages
+and provide a new shell with packages available.
+
+This is a great way to play with Nix tooling and see some of its potential.
+
+
+When are such environments useful?
+----------------------------------
+
+Sometimes you'd like **to use a tool, but it's not installed yet**. You don't want to
+bother installing software, you only want to have it now.
+
+Sometimes you'd like **to try a tool for a few minutes**. For example, there's a new shiny
+tool for writing presentation slides. 
+
+Sometimes you'd like **to give someone a one-liner also with instructions how to install
+tools** being used and such that it works between all Linux distributions and on MacOS.
+
+Sometimes you'd like **to provide a script that is reproducible**, meaning it will also provide the tooling.
+
+Searching package attribute names
```

Comment:
Some didactic wording for this section:

“What can I put in my shell environment?”

“To start, anything that's in the official package list can become part of the shell environment.”

“You can search the package list using…”

“Once you are comfortable doing this, you can add other things too. For example, packages of your own or custom shell aliases.”